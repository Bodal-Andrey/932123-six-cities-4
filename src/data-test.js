export default [
  {
    id: 1,
    city: `City-1`,
    title: `Title-1`,
    photo: `img-1`,
    price: 120,
    type: `Type-1`,
    rating: 4,
    isPremium: true,
    isBookmark: false,
    images: [`img-2`, `img-3`],
    description: [`description`],
    feature: `bedrooms`,
    guests: `adults`,
    inside: [`inside-1`, `inside-2`],
    owner: {
      avatar: `img-4`,
      name: `name`,
      isStar: true,
    }
  },
  {
    id: 2,
    city: `City-2`,
    title: `Title-2`,
    photo: `img-1`,
    price: 80,
    type: `Type-2`,
    rating: 4,
    isPremium: false,
    isBookmark: true,
    images: [`img-2`, `img-3`],
    description: [`description`],
    feature: `bedrooms`,
    guests: `adults`,
    inside: [`inside-1`, `inside-2`],
    owner: {
      avatar: `img-4`,
      name: `name`,
      isStar: false,
    }
  },
  {
    id: 3,
    city: `City-3`,
    title: `Title-3`,
    photo: `img-1`,
    price: 132,
    type: `Type-3`,
    rating: 4,
    isPremium: false,
    isBookmark: false,
    images: [`img-2`, `img-3`],
    description: [`description`],
    feature: `bedrooms`,
    guests: `adults`,
    inside: [`inside-1`, `inside-2`],
    owner: {
      avatar: `img-4`,
      name: `name`,
      isStar: false,
    }
  },
  {
    id: 4,
    city: `City-4`,
    title: `Title-4`,
    photo: `img-1`,
    price: 180,
    type: `Type-4`,
    rating: 5,
    isPremium: true,
    isBookmark: false,
    images: [`img-2`, `img-3`],
    description: [`description`],
    feature: `bedrooms`,
    guests: `adults`,
    inside: [`inside-1`, `inside-2`],
    owner: {
      avatar: `img-4`,
      name: `name`,
      isStar: true,
    }
  },
];
