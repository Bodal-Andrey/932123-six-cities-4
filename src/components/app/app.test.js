import React from "react";
import renderer from "react-test-renderer";
import App from "./app.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {offers} from "../../data-test.js";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

it(`App component test`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      offers,
      city: offers[0].city.name,
    },
    [NameSpace.APP]: {
      sortType: `popular`,
      activeOfferId: -1,
      showAuthPage: false,
    },
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`,
      authInfo: {
        avatarUrl: ``,
        name: ``,
        id: null,
        isPro: null,
        email: ``,
      }
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
      {
        createNodeMock: () => {
          return document.createElement(`div`);
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
