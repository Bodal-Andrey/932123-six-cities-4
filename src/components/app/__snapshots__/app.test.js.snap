// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test 1`] = `null`;
